.github/workflows/mips-build.yml

```diff
@@ -0,0 +1,55 @@
+name: MIPS Build and Package
+
+on:
+  workflow_dispatch:
+  push:
+    paths:
+      - 'o1vm/resources/programs/mips/**'
+      - 'Makefile'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust_toolchain_version: ["1.74"]
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        submodules: 'recursive'
+
+    - name: Cache apt packages
+      id: apt-cache
+      uses: actions/cache@v4
+      with:
+        path: |
+          /var/cache/apt/archives/*.deb
+        key: ${{ runner.os }}-apt-${{ hashFiles('.github/workflows/mips-build.yml') }}
+
+    - name: Install MIPS tools
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y binutils-mips-linux-gnu
+
+    - name: Build MIPS programs
+      run: make build-mips-programs
+
+    - name: Use shared Rust toolchain setting up steps
+      uses: ./.github/actions/toolchain-shared
+      with:
+        rust_toolchain_version: ${{ matrix.rust_toolchain_version }}
+
+    - name: Test elf_loader against mips programs
+      run: ./o1vm/test-gen-state-json.sh
+
+    - name: Create tar archive
+      run: |
+        cd o1vm/resources/programs/mips
+        tar -czf mips-binaries.tar.gz bin/
+
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: mips-binaries
+        path: o1vm/resources/programs/mips/mips-binaries.tar.gz
```

Comment:
Not blocking, but would be good to have it in a follow-up if not in this PR:
I would add a line in the README of o1vm mentioning how to download and run the test.